analyses/chromosomal-instability/01-localization-of-breakpoints.Rmd

```diff
@@ -0,0 +1,279 @@
+```
+
+Set up metadata
+
+```{r}
+# Read in the metadata
+metadata <- readr::read_tsv(file.path(data_dir, "pbta-histologies.tsv")) %>% 
+  # Only keep metadata for the samples we are working with here
+  dplyr::filter(Kids_First_Biospecimen_ID %in% common_samples)
+```
+
+Set up chromosome size data that will be used for creating genome bins.
+It just so happens that this BED file: `WGS.hg38.strelka2.unpadded.bed` is actually 
+just a list of the chromosome sizes so we are using that for now. 
+
+```{r}
+# Set up Chr sizes
+chr_sizes_bed <- readr::read_tsv(file.path(data_dir, "WGS.hg38.strelka2.unpadded.bed"),
+  col_names = c("chrom", "start", "end")
+) %>%
+  # Reformat the chromosome variable to drop the "chr"
+  dplyr::mutate(chrom = factor(gsub("chr", "", chrom),
+    levels = c(1:22, "X", "Y", "M")
+  )) %>%
+  # Remove sex chromosomes
+  dplyr::filter(!(chrom %in% c("X", "Y", "M")))
+
+# Make chromosome size named vector for Heatmap annotation
+chr_sizes_vector <- chr_sizes_bed$end
+names(chr_sizes_vector) <- chr_sizes_bed$chrom
+```
+
+Set up uncallable regions BED. We will use this to declare NAs in regions with
+too high of a percentage of uncallable regions. 
+
+```{r}
+uncallable_bed <- readr::read_tsv(
+  file.path(
+    "..",
+    "copy_number_consensus_call",
+    "ref",
+    "cnv_excluded_regions.bed"
+  ),
+  col_names = c("chrom", "start", "end")
+) %>%
+  # Reformat the chromosome variable to drop the "chr"
+  dplyr::mutate(chrom = factor(gsub("chr", "", chrom),
+    levels = c(1:22, "X", "Y")
+  )) %>%
+  dplyr::filter(
+    !is.na(chrom),
+    # Drop sex chromosomes
+    !(chrom %in% c("X", "Y", "M"))
+  )
+```
+
+### Mapping of breakpoint densities for individual samples 
+
+Let's reformat that the intersection_or_breaks data.frame. 
+It contains both SV data and CNV data for its intersections. 
+We'll go by SV data here. 
+
+```{r}
+# For simplicity's sake, change the name of the intersection column we want to use
+breaks_list$intersection_of_breaks <- breaks_list$intersection_of_breaks %>%
+  dplyr::rename(
+    chrom = sv_ranges.seqnames,
+    coord = sv_ranges.start
+  )
+```
+
+Calculate sample breakpoint densities across the binned genome.
+
+```{r}
+# Get a big list of break densities for each sample.
+sample_densities <- lapply(common_samples, function(sample_id) {
+  all_break_density(breaks_list = breaks_list, 
+                    sample_id = sample_id, 
+                    return_vector = TRUE) # We only want the total counts, 
+                                          # not the whole GenomicRanges object
+})
+
+# Bring along the sample IDs
+names(sample_densities) <- common_samples
+```
+
+Turn each dataset into its own data.frame and write it to its own TSV file. 
+This will be used by `02-a-plot-chr-instability-heatmaps.Rmd` for heatmaps. 
+
+```{r}
+# Turn intersection data into its own data.frame
+intersection_df <- purrr::map(sample_densities, "intersection_of_breaks") %>% 
```

Comment:
I found this confusing, until I found this note on the purrr cheat sheet.
![Screen Shot 2020-02-05 at 14 11 09 ](https://user-images.githubusercontent.com/20483/73874492-9c341100-4821-11ea-9850-f528a61481be.png)